website/integrations: mastodon: set correct uid field

dominic-r opened the pull request:
Setting the `OIDC_UID_FIELD` as `sub` created users on Mastodon with random usernames. Once this was changed to `preferred_username`, new users were created with their usernames set in authentik. My assumption is that users would wish to have the same username rather than have a random one generated.

<!--
👋 Hi there! Welcome.

Please check the Contributing guidelines: https://docs.goauthentik.io/docs/developer-docs/#how-can-i-contribute
-->

## Details

<!--
Explain what this PR changes, what the rationale behind the change is, if any new requirements are introduced or any breaking changes caused by this PR.

Ideally also link an Issue for context that this PR will close using `closes #`
-->
REPLACE ME

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

Commit:
website/integrations: mastodon: set correct uid field

Setting the `OIDC_UID_FIELD` as `sub` created users on Mastodon with random usernames. Once this was changed to `preferred_username`, new users were created with their usernames set in authentik. My assumption is that users would wish to have the same username rather than have a random one generated.

Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Merge branch 'goauthentik:main' into patch-2

BeryJu reviewed (commented):
👍 but there should be a warning that https://docs.goauthentik.io/docs/sys-mgmt/settings#allow-users-to-change-username should be set to false (which is the default but still) when using preferred_username as the primary identifier

tanberry commented on website/integrations/services/mastodon/index.md:
```diff
@@ -48,7 +48,7 @@ OIDC_DISCOVERY=true
 OIDC_AUTH_ENDPOINT=https://authentik.company/application/o/authorize/
 OIDC_SCOPE=openid,profile,email
-OIDC_UID_FIELD=sub
+OIDC_UID_FIELD=preferred_username
```
I thin it would be nice to explain bpth options to the readers. Maybe make `preferred_username` be the one in this list, but then below the list you could add a sentence to explain that one can use `OIDC_UID_FIELD=sub` if you want names generated?

tanberry reviewed (commented):
a suggestion... instead of removing info, we can add info (by showing both).

Commit:
website/integrations: mastodon: apply review suggestions

A: https://github.com/goauthentik/authentik/pull/11945#pullrequestreview-2427160470
B: https://github.com/goauthentik/authentik/pull/11945#discussion_r1837134142

Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Merge branch 'goauthentik:main' into patch-2

Commit:
Merge branch 'goauthentik:main' into patch-2

tanberry reviewed (approved):
Thanks, sorry it took me so long to get to this!
